operational/azure/azure_migrate/risc-azure-migrate.pt

```diff
@@ -0,0 +1,309 @@
+  end
+
+  result do
+    encoding "json"
+    collect jmes_path(response, "assets[*]") do
+      field "confirmed", jmes_path(col_item, "confirmed")
+      field "licensed", jmes_path(col_item, "licensed")
+      field "num_stack_members", jmes_path(col_item, "num_stack_members")
+      field "stack_name", jmes_path(col_item, "stack_name")
+      field "stackid", jmes_path(col_item, "stackid")
+      field "tags", jmes_path(col_item, "tags")
+    end
+  end
+end
+
+script "js_riscgetStackSummary", type: "javascript" do
+  result "request"
+  parameters "ds_riscauth", "param_acode"
+
+  code <<-EOS
+    
+    var authtoken = ds_riscauth.token
+    request = {
+      "host": "api.riscnetworks.com",
+      "verb": "GET",
+      "path": "/1_0/stacks/getSummary",
+      "headers": {
+        "User-Agent": "RS Policies",
+        "Content-Type": "application/json",
+        "assessmentcode": param_acode,
+        "token":  authtoken 
+      }
+    }
+
+  EOS
+end
+
+datasource "ds_riscgetAssetsByStack" do
+  iterate $ds_riscgetStackSummary
+  request do
+    run_script $js_riscgetAssetsByStack, $ds_riscauth, $param_acode, val(iter_item,"stackid")
+  end
+
+  result do
+    encoding "json"
+    collect jmes_path(response, "assets[*]") do
+      field "hostname", jmes_path(col_item, "data.hostname")
+      field "hardware_serial", jmes_path(col_item,"data.hardware_serial")
+      field "deviceid", jmes_path(col_item, "data.deviceid")
+      field "identifying_ip", jmes_path(col_item, "data.identifying_ip")
+      field "identifying_mac", jmes_path(col_item, "data.identifying_mac")
+      field "ostype", jmes_path(col_item, "data.os")
+      field "osversion", jmes_path(col_item, "data.os_version")
+      field "tags" do 
+        collect jmes_path(col_item, "tags[*]") do 
+          field "key", jmes_path(col_item, "tagkey")
+          field "value", jmes_path(col_item, "tagvalue")
+        end
+      end
+
+    end
+  end
+end
+
+script "js_riscgetAssetsByStack", type: "javascript" do
+  result "request"
+  parameters "ds_riscauth", "param_acode", "stackid"
+  code <<-EOS
+    var authtoken = ds_riscauth.token
+    request = {
+      "host": "api.riscnetworks.com",
+      "verb": "GET",
+      "path": "/1_0/assets/getAssets/byStack/"+ stackid,
+      "headers": {
+        "User-Agent": "RS Policies",
+        "Content-Type": "application/json",
+        "assessmentcode": param_acode,
+        "token":  authtoken
+      }
+    }
+
+  EOS
+end
+
+policy "discovered_devices" do
+  validate $ds_riscgetAssetsByStack do
+    summary_template "Discovered Servers to be Added to Azure Migrate"
+    detail_template <<-EOS
+# Discovered Servers to be Added to Azure Migrate
+| Hostname | IP Address | 
+| -------- | ---------- |
+{{ range data -}}
+| {{.hostname}} | {{.identifying_ip}}
+{{ end -}}
+EOS
+    escalate $esc_azure_migrate
+    check false
+  end
+end
+
+escalation "esc_azure_migrate" do
```

Comment:
approval should be added

Reply:
the purpose of this policy is really not to alert.  it's an integration whose sole purpose is to push all data from a RISC Foundations Assessment to Azure Migrate for migration planning.  approval defeats the purpose.